base/int.jl

```diff
@@ -552,10 +559,57 @@ end
 
 # checked ops are broken for 128-bit types (LLVM bug) ## FIXME: #4905
 
-checked_add(x::Int128, y::Int128) = x + y
-checked_sub(x::Int128, y::Int128) = x - y
-checked_mul(x::Int128, y::Int128) = x * y
+function checked_add(x::Int128, y::Int128)
+    # # x + y > typemax(Int128)
+    # y >= 0 && x > typemax(Int128) - y && throw(OverflowError())
+    # # x + y < typemin(Int128)
+    # y < 0 && x < typemin(Int128) - y && throw(OverflowError())
+    # x + y
+    r = x + y
+    # x and y have the same sign, and the result has a different sign
+    x$y >= 0 && x$r < 0 && throw(OverflowError())
```

Comment:
any idea how that compares efficiency wise to the more literal transformation of the comment `(x > 0 == y > 0) != (x > 0 == r > 0)`?

(we also may need to be aware that this comparison on the result may cause undefined behavior in llvm -- just like C -- although i think currently we aren't setting the NSW flag)